src/ensembl/src/content/app/species-selector/state/speciesSelectorActions.ts

```diff
@@ -158,42 +195,100 @@ export const handleSelectedSpecies: ActionCreator<
   dispatch(fetchAssemblies(genome_id));
 };
 
-export const commitSelectedSpecies = createStandardAction(
-  'species_selector/commit_selected_species'
-)();
+export const updateCommittedSpecies = createStandardAction(
+  'species_selector/toggle_species_use'
+)<CommittedItem[]>();
 
 export const commitSelectedSpeciesAndSave: ActionCreator<
   ThunkAction<void, any, null, Action<string>>
 > = () => (dispatch, getState) => {
-  dispatch(commitSelectedSpecies());
   const committedSpecies = getCommittedSpecies(getState());
+  const selectedItem = getSelectedItem(getState());
+
+  if (!selectedItem) {
+    return;
+  }
+
+  const newCommittedSpecies = [
+    ...committedSpecies,
+    buildCommittedItem(selectedItem)
+  ];
+
+  const speciesName = selectedItem.common_name || selectedItem.scientific_name;
+
+  analyticsTracking.trackEvent({
+    category: categories.ADD_SPECIES,
+    label: speciesName,
+    action: 'select'
+  });
+
+  dispatch(updateCommittedSpecies(newCommittedSpecies));
+  dispatch(clearSelectedSearchResult());
+
   speciesSelectorStorageService.saveSelectedSpecies(committedSpecies);
 };
 
-export const toggleSpeciesUse = createStandardAction(
-  'species_selector/toggle_species_use'
-)<string>();
-
 export const toggleSpeciesUseAndSave: ActionCreator<
   ThunkAction<void, any, null, Action<string>>
 > = (genomeId: string) => (dispatch, getState) => {
-  dispatch(toggleSpeciesUse(genomeId));
   const committedSpecies = getCommittedSpecies(getState());
+  let speciesName = '';
+  let currentStatus = '';
+  committedSpecies.map((item) => {
+    if (item.genome_id === genomeId) {
+      speciesName = item.common_name || item.scientific_name;
+      currentStatus = item.isEnabled ? 'do_not_use' : 'use';
+      return {
+        ...item,
+        isEnabled: !item.isEnabled
+      };
+    }
+  });
+
+  analyticsTracking.trackEvent({
+    category: categories.SELECTED_SPECIES,
+    label: speciesName,
+    action: currentStatus
+  });
+
+  dispatch(updateCommittedSpecies(committedSpecies));
   speciesSelectorStorageService.saveSelectedSpecies(committedSpecies);
 };
 
-export const deleteSpecies = createStandardAction(
-  'species_selector/delete_species'
-)<string>();
-
 export const deleteSpeciesAndSave: ActionCreator<
   ThunkAction<void, any, null, Action<string>>
 > = (genomeId: string) => (dispatch, getState) => {
-  dispatch(deleteSpecies(genomeId));
-  const committedSpecies = getCommittedSpecies(getState());
+  let speciesName = '';
+  const committedSpecies = getCommittedSpecies(getState()).filter((item) => {
+    if (item.genome_id === genomeId) {
+      speciesName = item.common_name || item.scientific_name;
+      return false;
+    }
+    return true;
+  });
```

Comment:
To avoid doing side effects outside of the filter function (which is not something one would expect), could we do:

```
const committedSpecies = getCommittedSpecies(getState())
const deletedSpecies = find(committedSpecies, ({genome_id}) => genome_id === genomeId);
const deletedSpeciesName = deletedSpecies.common_name || deletedSpecies.scientific_name;
const updatedCommittedSpecies = committedSpecies.filter(({genome_id}) => genome_id !== genomeId);

...do remaining stuff
```

Reply:
That's done